midterm_project/midterm-project - Eva - Ionut/muffin_man_shopping_list.py

```diff
@@ -0,0 +1,350 @@
+            print(f'Action Log: {product} quantity has been updated with {quantity} in the Fridge.'
+                  f'You now have {self.products_fridge[product]}')
+        else:
+            self.products_fridge[product] = quantity
+            print(f'Action Log: You did not have {product} in the Fridge. Quantity now updated to {quantity}')
+
+    def check_recipe(self, recipe):
+        items_to_buy = []
+        items_in_fridge = []
+        for ingredient in recipe.ingredients.keys():
+            if ingredient.title() in self.products_fridge.keys():
+                items_in_fridge.append(ingredient.title())
+            else:
+                items_to_buy.append(ingredient.title())
+        if items_to_buy and items_in_fridge:
+            print(f'Action Log: For the recipe {recipe.name} you already have {items_in_fridge}. '
+                  f'You need to buy {items_to_buy}')
+        elif not items_to_buy and items_in_fridge:
+            print(f'Action Log: You have all the ingredients for the {recipe.name} recipe: {items_in_fridge}')
+        else:
+            print(f'Action Log: You have none of the ingredients for the {recipe.name} recipe. '
+                  f'Go shopping and get {items_to_buy}')
+        return items_in_fridge, items_to_buy
+
+
+def check_the_fridge(fridge, recipe_box):
+    contents_fridge = set([item.lower() for item in fridge])
+    list_ingredients_for_recipe = {recipe: set(recipe_box[recipe].keys()) for recipe in recipe_box}
+
+    possible_recipes = []
+
+    for name, ingredients in list_ingredients_for_recipe.items():
+        if len(ingredients.intersection(contents_fridge)) >= len(ingredients) / 2:
+            possible_recipes.append(name)
+
+    print(f'Action Log: Possible recipes you can make with the products in your fridge:{(list(possible_recipes))}')
+    return possible_recipes
+
+
+def pretty_print_recipe(shop_list):
+    def pretty_recipe(*args):
+        if shop_list(*args):
+            muffin_man_string = r'''
+ ........................................................
+ :   ,-.      ,-.      ,-.      ,-.      ,-.      ,-.   :
+ : _(*_*)_  _(*_*)_  _(*_*)_  _(*_*)_  _(*_*)_  _(*_*)_ :
+ :(_  o  _)(_  o  _)(_  o  _)(_  o  _)(_  o  _)(_  o  _):
+ :  / o \    / o \    / o \    / o \    / o \    / o \  :
+ : (_/ \_)  (_/ \_)  (_/ \_)  (_/ \_)  (_/ \_)  (_/ \_) :
+ :   ,-.   ....................................   ,-.   :
+ : _(*_*)_ :                                  : _(*_*)_ :
+ :(_  o  _):          Shopping list:          :(_  o  _):
+ :  / o \  :                                  :  / o \  :
+ : (_/ \_) :                                  : (_/ \_) :'''
+
+            list_shop = shop_list(*args)
+            muffin_index = 1
+            space_number = 46
+            for index, (key, value) in enumerate(list_shop.items(), start=1):
+                if muffin_index == 1:
+                    muffin_string = ' :   ,-.   :   ' + str(index) + '. ' + key.title() + ': ' + str(value)
```

Comment:
aici poti scpa de mult cod daca faci o functie care iti returneaza muffin_man_string in functie de index, ceva de genu

```
def get_muggin_man_string(index):
    index_mapping = { 1: ' :   ,-.   :   ', 2:' : _(*_*)_ :   ', 3:' :(_  o  _):   ', 4:' :  / o \  :   ', 5:' : (_/ \_) :   '}
    muffin_string = index_mapping[index] + str(index) + '. ' + key.title() + ': ' + str(value)
    muffin_man_string += '\n' + muffin_string + ' ' * (space_number - len(muffin_string)) +  index_mapping[index] 
    return muffin_man_string
```
cred ca poti integra in functia de mai sus si urmatoarea serie de `if-uri` unde folosesti tot indexul